platform/spire/src/infra.py

```diff
@@ -8,11 +8,24 @@
 def admit(server_principal: str) -> str:
     config = configuration.get_config()
     principal_hostname = config.get_fqdn(server_principal)
-    if config.is_kerberos_enabled():
-        return access.call_keyreq("bootstrap-token", principal_hostname).decode().strip()
-    else:
+
+    errs = []
+
+    try:
+        if config.is_kerberos_enabled():
+            return access.call_keyreq("bootstrap-token", principal_hostname).decode().strip()
+    except Exception as e:
+        errs.append(e)
```

Comment:
This has the same problem as before, where errors are hidden, but it doesn't look like this even notifies the user that it had to resort to keyinitadmit instead of keyreq.